File: demo/src/main/java/com/example/demo/NoticeService.java
package com.example.demo;

import org.springframework.stereotype.Service;
import java.time.LocalDate;
import java.util.List;

@Service
public class NoticeService {

    private final NoticeRepository noticeRepository;

    public NoticeService(NoticeRepository noticeRepository) {
        this.noticeRepository = noticeRepository;
    }

    // Post a notice
    public Notice postNotice(Notice notice) {
        notice.setPostedAt(LocalDate.now());
        return noticeRepository.save(notice);
    }

    // Get all notices
    public List<Notice> getAllNotices() {
        return noticeRepository.findAll();
    }

    // Get one notice
    public Notice getNoticeById(Long id) {
        return noticeRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Notice not found"));
    }

    // Delete a notice
    public void deleteNotice(Long id) {
        noticeRepository.deleteById(id);
    }

}

File: frontend/src/services/noticeService.js
import API from './api';

export const getNotices = async () => {
  const response = await API.get('/notices');
  return response.data;
};

export const addNotice = async (notice) => {
  const response = await API.post('/notices', notice);
  return response.data;
};

export const deleteNotice = async (id) => {
  const response = await API.delete(`/notices/${id}`);
  return response.data;
};